app/renderer/actions/Inspector.js

```diff
@@ -137,26 +137,24 @@ export function selectElement (path) {
   return async (dispatch, getState) => {
     // Set the selected element in the source tree
     dispatch({type: SELECT_ELEMENT, path});
-    const state = getState().inspector;
-    const {attributes: selectedElementAttributes, xpath: selectedElementXPath} = state.selectedElement;
-    const {sourceXML} = state;
+    const { selectedElement, sourceXML, expandedPaths } = getState().inspector;
 
     // Expand all of this element's ancestors so that it's visible in the source tree
-    let {expandedPaths} = getState().inspector;
+    const copiedExpandedPaths = expandedPaths.slice();
```

Comment:
I usually use [...arr] to copy an array

Reply:
it also makes sense to add a comment about why it is necessary to copy the array

Reply:
Perhaps eventually it would make sense to start using https://github.com/immutable-js/immutable-js in this project to avoid similar issues

Reply:
Yeah, agreed. I believe React recommends using Immer for this purpose: https://immerjs.github.io/immer/